packages/nuxt/src/pages/runtime/plugins/router.ts

```diff
@@ -61,9 +62,17 @@ const plugin: Plugin<{ router: Router }> = defineNuxtPlugin({
 
     const routes = routerOptions.routes?.(_routes) ?? _routes
 
+    let startPosition: Parameters<RouterScrollBehavior>[2] | null
     const initialURL = process.server ? nuxtApp.ssrContext!.url : createCurrentLocation(routerBase, window.location)
     const router = createRouter({
       ...routerOptions,
+      scrollBehavior: (to, from, savedPosition) => {
+        if (from === START_LOCATION) {
+          startPosition = savedPosition
+          return
+        }
+        return routerOptions.scrollBehavior?.(to, from, nuxtApp.isHydrating ? startPosition : savedPosition)
```

Comment:
I don't know which one is better: using `nuxtApp.isHydrating` or using `startPosition` only once and then using saved position. It seems like you reset it anyway in line 206. Otherwise this looks good